### PR TITLE
Further clamped Character Limits on AI Law Boards

### PR DESCRIPTION
**Pull request by MarkNais**
## About the PR 
default limit is set to ~~50~~ 100 chars
Freeform is set to ~~200~~ 400 chars
(Old limit was 1024 for all)

## Why's this needed? 
Abuse of non-freeform boards.

## Changelog

```changelog
(u)MarkNstein
(*)AI Law Boards now have tighter character limits.
```
[BALANCE]

### CODE REVIEW

**Comment by UrsulaMejor** [comment]
This string has 50 characters, for your reference.

**Comment by UrsulaMejor** [comment]
I think that 50 characters is too small for the default. It makes it pretty much impossible to do any cool shit with the non-freeform laws, and makes ion storm laws essentially impossible to deal with except by killing the AI.

**Comment by mordent-goonstation** [comment]
> I think that 50 characters is too small for the default. It makes it pretty much impossible to do any cool shit with the non-freeform laws, and makes ion storm laws essentially impossible to deal with except by killing the AI.

I would prefer the ion storm issue was solved by having a method to "fix" Laws 1-3.

**Comment by penny64** [comment]
It's not abuse to use non-freeform modules creatively I'm certain this is valid behavior and as designed- and more importantly, funny and fun

**Comment by BatElite** [comment]
If that's about the length of 100 characters, then 200 for the freeform is way tiny as well. "This law overrides all other laws" is already 33, and that's basically traitor boilerplate. I'd think something closer to 500 so people can actually write stuff with some nuance or clarifying clauses.

**Comment by Merlin1230** [comment]
I don’t think the AI modules really need this small of a character limit. I haven’t seen it been abused, unless writing really long laws count as abuse.

**Comment by flrsh** [comment]
I agree with other commenters; maybe 100 for default and 300-500 for freeform?

File: code/obj/item/ai_modules.dm [hunk 1]
```diff
@@ -20,6 +20,7 @@ AI MODULES
 	throw_speed = 3
 	throw_range = 15
 	mats = 8
+	var/input_char_limit = 50
```
Comment (MarkNais) [inline code/obj/item/ai_modules.dm:23]:
```suggestion
	var/input_char_limit = 100
```

File: code/obj/item/ai_modules.dm [hunk 2]
```diff
@@ -241,6 +242,7 @@ AI MODULES
 /obj/item/aiModule/freeform
 	name = "'Freeform' AI Module"
 	lawNumber = 14
+	input_char_limit = 200
```
Comment (MarkNais) [inline code/obj/item/ai_modules.dm:245]:
```suggestion
	input_char_limit = 400
```

**Review by MarkNais — COMMENTED** [review]
Per feedback: doubling

**Comment by PrimeNumb** [comment]
It's worth noting that there'll be an inconsistency with how many characters the input box itself can take and how many characters are actually saved to the module. Might be a good idea to use TGUI and clamp the input value in the UI as well to better communicate to the user where exactly it starts chopping the text off.

**Comment by MarkNais** [comment]
> It's worth noting that there'll be an inconsistency with how many characters the input box itself can take and how many characters are actually saved to the module. Might be a good idea to use TGUI and clamp the input value in the UI as well to better communicate to the user where exactly it starts chopping the text off.

I've already worked 50hrs this week so fat chance I spend any real effort on this.
However the pre-existing behavior is that after confirming your entry, it repeats the whole board's contents to you, so users can double-check the end result.

**Comment by PrimeNumb** [comment]
> > It's worth noting that there'll be an inconsistency with how many characters the input box itself can take and how many characters are actually saved to the module. Might be a good idea to use TGUI and clamp the input value in the UI as well to better communicate to the user where exactly it starts chopping the text off.
> 
> I've already worked 50hrs this week so fat chance I spend any real effort on this.
> However the pre-existing behavior is that after confirming your entry, it repeats the whole board's contents to you, so users can double-check the end result.

Oh absolutely, I understand. I just wanted to throw that out there that this should probably be done at some point (whether now or later). The idea of a tighter input limit (the focus of this PR) is a sound one, in my opinion.

**Review by flrsh — APPROVED** [review]

**Pull request merged**